### PR TITLE
Add template for AclProviderInterface

### PR DESCRIPTION
**Pull request by VincentLanglet**
Hi @ondrejmirtes 

If I understand correctly template and I didn't make any mistake, this PR seems better than https://github.com/symfony/security-acl/pull/60.

By the way I have two others questions:
- WDYT about supporting directory in the `stubFiles` config ? This way the following code would be enough.
```
stubFiles:
    - stubs/
```

- WDYT about using the same directory structure than Symfony inside the stub directory ? There will be more and more file, and it can be easier to find them if we follow the namespace.

### CODE REVIEW

File: stubs/AclProviderInterface.stub [hunk 1]
```diff
@@ -0,0 +1,21 @@
+<?php
+
+namespace Symfony\Component\Security\Acl\Model;
+
+use Symfony\Component\Security\Acl\Model\AclInterface;
+
+/**
+ * @template TAcl of AclInterface
+ */
+interface AclProviderInterface extends AclProviderInterface
```
Comment (ondrejmirtes) [inline stubs/AclProviderInterface.stub:10]: `AclProviderInterface extends AclProviderInterface`?

Comment (ondrejmirtes) [reply stubs/AclProviderInterface.stub:10]: And BTW you don't have to solve this with generics, I'm not sure it makes sense here. You can simply override the method in the child MutableAclProviderInterface and say it returns MutableAclInterface.

Comment (VincentLanglet) [reply stubs/AclProviderInterface.stub]: > And BTW you don't have to solve this with generics, I'm not sure it makes sense here. You can simply override the method in the child MutableAclProviderInterface and say it returns MutableAclInterface.

I made a new try.

Comment (ondrejmirtes) [reply stubs/AclProviderInterface.stub]: Yeah, this is nicer, thanks. Please fix the build, you need to follow this tidbit:

> Why is PHPStan complaining about “Class not found” in a stub file?

https://phpstan.org/user-guide/stub-files

Comment (ondrejmirtes) [reply stubs/AclProviderInterface.stub]: Yeah, namespaces could be honored in the directory structure :)

For the directories it'd have to be a new config key `stubDirectories` but I don't think it's really necessary.

Comment (VincentLanglet) [reply stubs/AclProviderInterface.stub]: I made the following changes:
- honored namespaces with directory inside the stubs one
- run the linter on the stubs files.
- Add the missing stubs classes.

Not every methods are defined in a stubs file. In the same way, do I have to write `interface MutableAclProviderInterface extends AclProviderInterface` in a stub file or can I just write 
`interface MutableAclProviderInterface` and avoid defining `AclProviderInterface` ?

Comment (ondrejmirtes) [reply stubs/AclProviderInterface.stub]: The build is still failing. It's better to write `interface MutableAclProviderInterface extends AclProviderInterface`, seems more complete, although that information is currently unused from the stubs.

**Comment by VincentLanglet** [comment]
I need https://github.com/phpstan/phpstan/issues/3111 first

**Comment by VincentLanglet** [comment]
@ondrejmirtes I removed the `findAcls` annotation since it requires the next phpstan major.
So this PR is now ready to merge.

**Comment by VincentLanglet** [comment]
@ondrejmirtes I rebase master ; this PR is ready to merge

**Comment by ondrejmirtes** [comment]
Hi, I'm sorry, but this PR does too many things at once and it's unlikely to be merged.

1) Adds AclProviderInterface stub - should be in a separate PR.
2) Moves stubs into subdirectories - should be a separate PR.
3) Adds CS to stubs - not interested in this, sorry, as other repositories don't have this and there's little value in that.

**Pull request closed without merging**